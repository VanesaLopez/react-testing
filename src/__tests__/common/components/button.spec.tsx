
import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Button } from 'common/components/button';

describe('Button specs', () => {
  it('should render', () => {
    // Arrange
    const props = {
        label: 'label',
        onClick: jest.fn()
      };

    // Act
    const { getByText, asFragment } = render(<Button {...props}/>);

    // Assert
    getByText(props.label);
    expect(asFragment()).toMatchSnapshot();
    
  });

  it('should call onClick', () => {
    // Arrange
    const props = {
        label: 'label',
        onClick: jest.fn()
    };

    // Act
    const { getByText } = render(<Button {...props} />);

    const buttonElement = getByText(props.label);
    fireEvent.click(buttonElement);

    // Assert
    expect(props.onClick).toHaveBeenCalled();
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});