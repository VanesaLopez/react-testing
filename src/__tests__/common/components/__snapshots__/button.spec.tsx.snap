// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button specs should render 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
